### PR TITLE
Block party

### PR DESCRIPTION
**Pull request by nternetinspired**
# A premise, I promise

I think in order to facilitate ease-of-use, readability and maintenance related, and dependant, html should be rendered in one place. This is why we have layouts, after all. 
## Paints go in a paintbox, pencils in a pencil tin.

Currently, the interdependent markup for the article infoblock is scattered across 8 files. The parent block.php defines the content as a document list and each child element contains the document definition. 

The problem? a `<dd>` can not live a life independent of a `<dl>`, so these structural markup elements should live together.
### Once upon a time…

A young prince (not me) decided that his document information would be better served as a list of items and set out upon a great and noble quest. However, upon reaching the hitherto little-trod realm of `/layouts/joomla/content/info-block` this young prince did discover he must slay(override) not 1, but 8 fearsome dragons(layout files). This brought great sadness upon our hero and he was avowed to forge, with great cunning, a plan such that none should have to endure these trials again.

:hocho: :dragon: 
## TL:DR;

This PR pulls related markup from the 8 infoblock overrides into the parent, infoblock.php. This means that any wishing to change the markup, to an unordered list or simple paragraph for example, can do so with just one override.
### Testing

Apply patch and check that neither appearance or markup of article info block is changed.

### CODE REVIEW

**Comment by G-Lu** [comment]
Tested the patch on serveral templates and got no differnece in the appearance. (not patched/patched)
In some templates the html-code got some diffrent white spaces and line breaks but the essential code was the same.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4406">issues.joomla.org/joomla-cms/4406</a>.</sub>

**Comment by Blutengel89** [comment]
I have tested the patch, but i can´t use <dd> without <dl>.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4406">issues.joomla.org/joomla-cms/4406</a>.</sub>

**Comment by nternetinspired** [comment]
> Tested the patch on serveral templates and got no differnece in the appearance. (not patched/patched)
In some templates the html-code got some diffrent white spaces and line breaks but the essential code was the same.

There isn't supposed to be a difference in appearance.

> I have tested the patch, but i can´t use dd without dl.

Correct. What is your point?

**Comment by hans2103** [comment]
I have tested this item :white_check_mark: successfully on c76a74ed2fe1a9922f24e830a5b5d3fb9cbb55ed<br /><br />Great improvement!
Moving HTML to the correct place is well executed. 
Before & after patch the HTML output looks the same. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4406">issues.joomla.org/joomla-cms/4406</a>.</sub>

**Comment by slibbe** [comment]
I have tested this item :white_check_mark: successfully on c76a74ed2fe1a9922f24e830a5b5d3fb9cbb55ed<br /><br />I have tested this succesfully with no apparent diffence.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4406">issues.joomla.org/joomla-cms/4406</a>.</sub>

**Comment by nternetinspired** [comment]
Thanks for testing @hans2103 and @slibbe :)

**Comment by zero-24** [comment]
RTC based on tests. Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4406">issues.joomla.org/joomla-cms/4406</a>.</sub>

**Comment by Hackwar** [comment]
This change is not backwards compatible. If you overrode block.php, you will end up with missing dd-tags around your code. I also still would rather see this in one big file than in a gazillion tiny ones. There is NO gain at all by putting every single line of code in its own file, except that we unnecessarily inflate the codebase with comments and file headers. This also is becoming a performance problem.

We can not improve this in any way in 3.x, since moving it all in one big file would prevent overrides for one of the small files to work and changing the code in the files like in here breaks existing overrides.

Might be a good reason to attack 4.0 without trying to invent the great solution for everything. Keep 4.0 simple by simply droping the deprecated stuff from 3.x and fixing stuff like this one and then release that.

**Comment by zero-24** [comment]
Pending<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4406">issues.joomla.org/joomla-cms/4406</a>.</sub>

**Comment by roland-d** [comment]
@nternetinspired Thank you for your contribution. The PLT has decided to close this PR and not merge it in Joomla 3 but reconsider it for Joomla 4. We want to look at not scattering a page around in so many single files.
